Fix compile errors in builder when model disabled

FallenWarrior2k opened the pull request:
PR #1830 fixed a bunch of clippy pedantic level lints, which also unfortunately caused some regressions for the no-model use case, as I intentionally used bare Default in #1770 to automatically select the correct type without manually inspecting the feature flags again.

I admit my use case is somewhat outlandish, but would you mind if I added one or two regression tests to CI to avoid situations like this in the future?

FallenWarrior2k committed:
Fix compile errors in builder when model disabled

PR #1830 fixed a bunch of clippy pedantic level lints, which also
unfortunately caused some regressions for the no-model use case, as I
intentionally used bare Default in #1770 to automatically select the
correct type without manually inspecting the feature flags again.

FallenWarrior2k committed:
Fix typo in comment

arqunis commented:
> but would you mind if I added one or two regression tests to CI to avoid situations like this in the future

Sure, go ahead. Nick had no way of knowing it would break compilation of the library without the `model` feature. With new CI jobs, they should prevent that scenario from happening again.

FallenWarrior2k commented:
I'm not blaming them. Can't expect everybody to manually check for every combination of flags there is.

FallenWarrior2k committed:
Add CI job to test builder without model

FallenWarrior2k committed:
Add CI job to build unstable w/o default features

FallenWarrior2k commented:
Does this look good to you?

arqunis commented:
Looks good. If you handle the unused imports, I'll merge this.

FallenWarrior2k committed:
Fix unused imports in builder when model disabled

FallenWarrior2k commented:
On it.

FallenWarrior2k committed:
Disable testing builder without model

A ton of tests are failing to compile because they missing feature gates
for the functionality they're trying to test. Fixing all of those
exceeds the scope of this PR imo.

FallenWarrior2k commented:
As explained in the commit message, I disabled the tests for builder without model for now because a lot of them currently fail to compile due to missing feature gates. I would address that in another PR to prevent this one from blowing up in scope even further.
